FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.JobManagement.UnitTests/AzureStorageQueueClientTests.cs

```diff
@@ -0,0 +1,1273 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Azure;
+using Azure.Data.Tables;
+using Azure.Storage.Queues;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.Common.Models.Jobs;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Extensions;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models.AzureStorage;
+using Microsoft.Health.JobManagement;
+using Xunit;
+using JobStatus = Microsoft.Health.JobManagement.JobStatus;
+
+namespace Microsoft.Health.Fhir.Synapse.JobManagement.UnitTests
+{
+    /// <summary>
+    /// Use different queue type for integration test to avoid conflict
+    /// </summary>
+    internal enum TestQueueType : byte
+    {
+        GivenNewJobs_WhenEnqueueJobs_ThenCreatedJobsShouldBeReturned = 16,
```

Comment:
Maybe we can add enqueue test for table entity exist

Reply:
Sure, the following unit tests test different scenarios of re-enqueueing:

GivenCreatedJob_WhenEnqueueJobAgain_ThenTheExistingOneShouldBeReturned,
GivenRunningJob_WhenEnqueueJobAgain_ThenTheExistingOneShouldBeReturned,
GivenFinishedJob_WhenEnqueueJobAgain_ThenTheExistingOneShouldBeReturned,
GivenEnqueueFailed_WhenEnqueueJobAgain_ThenContinueToEnqueue,